src/libraries/apicompat/ApiCompat.proj

```diff
@@ -6,35 +6,40 @@
     <NoTargetsDoNotReferenceOutputAssemblies>false</NoTargetsDoNotReferenceOutputAssemblies>
     <!-- Target reference assemblies instead of implementation assemblies. -->
     <CompileUsingReferenceAssemblies>true</CompileUsingReferenceAssemblies>
-    <TrimOutPrivateAssembliesFromReferencePath>true</TrimOutPrivateAssembliesFromReferencePath>
 
     <ApiCompatNetCoreAppLatestStableBaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.NetCoreAppLatestStable.xml</ApiCompatNetCoreAppLatestStableBaselineFile>
     <ApiCompatNetStandard21BaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netstandard2.1.xml</ApiCompatNetStandard21BaselineFile>
     <ApiCompatNetStandard20BaselineFile>$(MSBuildThisFileDirectory)ApiCompatBaseline.netstandard2.0.xml</ApiCompatNetStandard20BaselineFile>
   </PropertyGroup>
 
+  <!-- In servicing, the live targeting pack is compared against the GA version in strict mode, without a suppression file. -->
+  <PropertyGroup Condition="'$(DotNetFinalVersionKind)' == 'servicing' and '$(ApiCompatNetCoreAppBaselineVersion)' == '$(NetCoreAppCurrentVersion)'">
+    <ApiCompatEnableStrictMode>true</ApiCompatEnableStrictMode>
+    <ApiCompatNetCoreAppLatestStableBaselineFile />
+  </PropertyGroup>
```

Comment:
It might be useful to change the condition here to be `Condition="'$(DotNetFinalVersionKind)' == 'servicing'"` and use that as the forcing function to make sure we update `ApiCompatNetCoreAppBaselineVersion`? Or do we have a doc somewhere that has all of the steps to do to make sure that the version gets updated?

Reply:
Agreed. Pushed a commit that changes the condition. I also just noticed that with https://github.com/dotnet/sdk/commit/a3303b614e235dea5d90283bb8fcc09d5bfcb86f, unnecessary suppressions will be flagged which is another good forcing function.

@carlossanlop has a servicing doc. We need to add the following to it:

When changing the repository into servicing mode, the following needs to happen for APICompat:
1. Update the `ApiCompatNetCoreAppBaselineVersion` property and point it to the latest available targeting pack version (the closest one to the GA version). Adjust the `ApiCompatNetCoreAppBaselineTFM` property as well (i.e. when switching from `7.0.0` to `8.0.0-rc2....`, the TFM changes from `net7.0` to `net8.0`).
2. APICompat will then complain about unnecessary suppressions in the baseline files. Remove all unnecessary files / entries from the suppression files.

With that, introducing a new API or breaking change in servicing is now guarded against.